lavalink/audio_events.py

```diff
@@ -92,16 +91,38 @@ def __init__(self, player, *args, **kwargs):
             await self.track_stuck(event)
 
 
+class InternalEventAdapter(AbstractPlayerEventAdapter):
+    async def track_pause(self, event: TrackPauseEvent):
+        pass
+
+    async def track_resume(self, event: TrackResumeEvent):
+        pass
+
+    async def track_start(self, event: TrackStartEvent):
+        pass
+
+    async def track_end(self, event: TrackEndEvent):
+        #if event.reason != 'REPLACED' and event.reason != 'STOPPED':
+        self.player.current = None
+
+    async def track_exception(self, event: TrackExceptionEvent):
+        pass
+
+    async def track_stuck(self, event: TrackStuckEvent):
+        pass
+
+
 class DefaultEventAdapter(AbstractPlayerEventAdapter):
     """
     The default event adapter
     TODO: Add more shit to this class
     """
+
     def __init__(self, player, ctx):
         super().__init__(player)
         self.ctx = ctx
         self.bot = ctx.bot
-        self.queue = deque()
```

Comment:
Deque probably would've been a better option to allow users to add something to the start of the queue?